File: solutions/c/darts/1/darts.c
#include "darts.h"

#include <math.h>

uint8_t score(coordinate_t pos)
{
    float distance = sqrt((pos.x * pos.x) + (pos.y * pos.y));

    if (distance <= INNER)
        return POINTS_10;
    else if (distance <= MIDDLE)
        return POINTS_5;
    else if (distance <= OUTER)
        return POINTS_1;
    
    return POINTS_0;
}

File: solutions/ruby/darts/1/darts.rb
class Darts
  POINTS = {
    outside: 0,
    outer: 1,
    middle: 5,
    inner: 10
  }

  DISTANCE = {
    outer: 100,
    middle: 25,
    inner: 1
  }
  
  attr_reader :score

  def initialize(x, y)
    distance = x**2 + y**2

    @score = if distance <= DISTANCE[:inner]
      POINTS[:inner]
    elsif distance <= DISTANCE[:middle]
      POINTS[:middle]
    elsif distance <= DISTANCE[:outer]
      POINTS[:outer]
    else
      POINTS[:outside]
    end
  end
end
    

    